airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+from copy import deepcopy
+from typing import Any, Dict, Optional
+
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.types import Config, StreamSlice, StreamState
+
+
+class NewtoLegacyFieldTransformation(RecordTransformation):
+    """
+    Implements a custom transformation which maps a "new" field to the legacy equivalent of the field for streams which contain Deals and Contacts entities.
+    """
+
+    def __init__(self, field_mapping: Dict[str, str]) -> None:
+        self._field_mapping = field_mapping
+
+    def transform(
+        self,
+        record: Dict[str, Any],
+        config: Optional[Config] = None,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+    ) -> Dict[str, Any]:
+        """
+        Transform a record by adding fields directly to the record by manipulating a "new" field into a legacy field to avoid breaking syncs.
```

Comment:
If I understand correctly from the test, it does not only transform a record but also schemas, right? Can we update this description and re-assign the variable `record` to a variable with a better name to make it clear like `record_or_schema`?

Reply:
Yes, I agree with that.